test/fixtures/non-string-argument/expected.js

```diff
@@ -0,0 +1,21 @@
+new Promise(resolve => {
+  resolve(require(new String({ 'answer': 42 })));
```

Comment:
Also I just tried this
```
npm i false
```
then created `test.js`
```
console.log(require(false));
```
when I run it I'm getting `AssertionError: missing path`

So it seems that `require` doesn't coerce its argument